src/fidesops/ops/api/v1/endpoints/consent_request_endpoints.py

```diff
@@ -0,0 +1,276 @@
+    provided_identity = _get_consent_request_and_provided_identity(
+        db=db, consent_request_id=consent_request_id, verification_code=data.code
+    )
+
+    if not provided_identity.encrypted_value:
+        raise HTTPException(
+            status_code=HTTP_404_NOT_FOUND, detail="Provided identity missing email"
+        )
+
+    consent = Consent.filter(
+        db=db, conditions=Consent.provided_identity_id == provided_identity.id
+    ).all()
+
+    identity = (
+        Identity(email=provided_identity.encrypted_value["value"])
+        if provided_identity.field_name == ProvidedIdentityType.email
+        else Identity(phone_number=provided_identity.encrypted_value["value"])
+    )
+
+    if not consent:
+        return ConsentPreferences(identity=identity, consent=None)
+
+    return ConsentPreferences(
+        identity=identity,
+        consent=[
+            Consent(
+                data_use=x.data_use,
+                data_use_description=x.data_use_description,
+                opt_in=x.opt_in,
+            )
+            for x in consent
+        ],
+    )
+
+
+@router.patch(
+    CONSENT_REQUEST_PREFERENCES,
+    status_code=HTTP_200_OK,
+    response_model=ConsentPreferences,
+)
+def set_consent_preferences(
+    *,
+    consent_request_id: str,
+    db: Session = Depends(get_db),
+    data: ConsentPreferencesWithVerificationCode,
+) -> ConsentPreferences:
+    """Verifies the verification code and saves the user's consent preferences if successful."""
+    if not data.consent:
+        raise HTTPException(
+            status_code=HTTP_400_BAD_REQUEST, detail="No consent preferences provided"
+        )
+
+    provided_identity = _get_consent_request_and_provided_identity(
+        db=db,
+        consent_request_id=consent_request_id,
+        verification_code=data.code,
+    )
+
+    if not provided_identity.encrypted_value:
+        raise HTTPException(
+            status_code=HTTP_404_NOT_FOUND, detail="Provided identity missing email"
+        )
+
+    for preference in data.consent:
+        current_preference = Consent.filter(
+            db=db,
+            conditions=(Consent.provided_identity_id == provided_identity.id)
+            & (Consent.data_use == preference.data_use),
+        ).first()
+
+        if current_preference:
+            current_preference.update(db, data=dict(preference))
+        else:
+            preference_dict = dict(preference)
+            preference_dict["provided_identity_id"] = provided_identity.id
+            Consent.create(db, data=preference_dict)
+
+    preferences = Consent.filter(
+        db, conditions=(Consent.provided_identity_id == provided_identity.id)
+    ).all()
+
+    if not preferences:
+        return ConsentPreferences(identity=data.identity, consent=None)
+
+    return ConsentPreferences(
+        identity=data.identity,
+        consent=[
+            ConsentSchema(
+                data_use=x.data_use,
```

Comment:
Is there any place we can limit what the data use is in code?

Reply:
I think the intention is for this to be limited to [Fideslang data uses](https://github.com/ethyca/fideslang/blob/main/src/fideslang/default_taxonomy.py#L570-L755)?

Reply:
Data uses can have custom values added right? If so can I validate against them since I won't have a way to know all values?

Reply:
OK we can leave validation off